README.md

```diff
@@ -1 +1,182 @@
+  - [Run Tests](#run-tests)
+  - [👥 Authors ](#-authors-)
+- [🔭 Future Features:](#-future-features)
+  - [🤝 Contributing ](#-contributing-)
+  - [⭐️ Show your support ](#️-show-your-support-)
+  - [🙏 Acknowledgments ](#-acknowledgments-)
+  - [📝 License ](#-license-)
+
+# MDoc <a name="about-project"></a>
+
+It's an Online Doctor Reservation web application designed to provide users with the ability to reserve, view, and manage doctors and doctor appointment reservation . The system aims to streamline the process of finding and booking doctors from various locations
+
+#### Link to [Frontend Repository](https://github.com/siddghosh108/m-doc)
+
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+<details>
+  <summary>Framework</summary>
+  <ul>
+    <li><a href="https://rubyonrails.org/">Ruby on Rails</a></li>
+    <li><a href="https://www.postgresql.org/">PostgreSQL</a></li>
+  </ul>
+</details>
+
+### Key Features <a name="key-features"></a>
+
+- User Access: The system allows users to create accounts and log in
+- Doctor reservation: hospitals can list their available doctors. Each listing includes details such as doctors name, date, and location.
+- Reservation Management: Users can view available doctors  at different locations and make reservations
+- Add and delete doctors in including location
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- <a href="https://capstone-m-doc.onrender.com/">Live Demo Link</a>
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🎬 Kanban Board <a id="kanban"></a>
+
+Link to the [Kanban Board](https://github.com/users/prg-04/projects/5)
+
+All 3 Authors have been contributing to this project from start to finish
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+### Prerequisites
+
+In order to run this project you need:
+
+- Installed Git.
+- Installed Ruby on Rails
+- Installed PostgreSQL
+- Create a local directory that you want to clone the repository.
+- Open the command prompt in the created directory.
+- On the terminal run this command git clone https://github.com/prg-04/MDoc.git
+
+### Setup
+
+Clone this repository to your desired folder:
+
+```sh
+git clone https://github.com/prg-04/MDoc.git
+```
+
+```sh
+cd MDoc
+```
+
+```sh
+bundle install
+```
+
+```sh
+1. Remove config/master.key and config/credentials.yml.enc if they exist.
+2. Run: `rails secret`.  Copy the key.
+3. Run: EDITOR="code --wait" bin/rails credentials:edit
+4. In the editor that opens, add this:  devise_jwt_secret_key: <the key you copied in step 2>
+5. Save the file and close the editor.
+
+New master.key, credentials.yml.enc files will be generated,
+and the key will be stored in `Rails.application.credentials.devise_jwt_secret_key`.
+```
+
+```sh
+rails db:create db:migrate db:seed
+```
+
+```sh
+rails s
+```
+
+### Tests
+
+After setup, you ran use the comman below to run RSpec
+
+```sh
+rspec
+```
```

Comment:
- [ ] After the changes, we got 3 failing specs:

![image](https://github.com/prg-04/MDoc/assets/57421823/4d436a8a-f8a0-4aaa-ab11-7e4a12cd5f0a)

## Explanation and solution

This is because, in all 3 controllers, the status for the #destroy action has been set to 200(ok), but all the 3 specs have been set to expect a status of 204(no_content).

Look:

doctors_controller.rb:

![image](https://github.com/prg-04/MDoc/assets/57421823/7e433dcd-a448-4381-994a-330691e5d1dc)

doctors_spec.rb

![image](https://github.com/prg-04/MDoc/assets/57421823/d0dc249b-5a70-4332-b6d9-bf5cc3502bda)

Debugging with pry:

![image](https://github.com/prg-04/MDoc/assets/57421823/5a67187f-046e-4af1-b5b9-9aa6bae72424)

### Therefore, just make the controllers and the tests expect the same status and all will pass (tested).


Now that you know the cause and also the solution, kindly make those tests pass so your app's integrity is secured. 👍